script.js

```diff
@@ -117,26 +114,85 @@ const deleteAttendance = ({ registerId, attendanceId }) => {
   console.log(`lesson with id: ${attendanceId} not found.`);
 }
 
+// Funzione per ottenere la lista degli studenti
+const getStudentList = () => {
+  return JSON.parse(localStorage.getItem("students"));
+};
+
+// Funzione per ottenere uno specifico studente
+const getStudent = (id) => {
+  const students = getStudentList();
+  const studentFound = students.find((student) => student.id === "" + id);
+
+  return studentFound;
+}
+
 // Funzione per creare uno studente
 const createStudent = ({ name, lastName, email, lectures }) => {
-  lastStudId = parseInt(lastStudId);
-  lastStudId++;
+  const prevStudents = getStudentList() || [];
 
-  name = normalizeName(name);
-  lastName = normalizeName(lastName);
+  const lastStudId = JSON.parse(localStorage.getItem("lastStudId")) || "0";
+  const newStudId = parseInt(lastStudId) + 1;
 
-  const sampleStudent = {
-    id: '' + lastStudId,
-    name: name,
-    lastName: lastName,
-    email: email,
-    lectures: []
-  }
+  const newStudents = [...prevStudents, {
+    id: '' + newStudId,
+    name: normalizeName(name),
+    lastName: normalizeName(lastName),
+    email,
+    lectures,
+  }];
+
+  console.log(`student ${JSON.stringify(newStudents.slice(-1)[0])} created successfully.`);
 
-  sampleStudent.lectures.push(...lectures);
+  localStorage.setItem("students", JSON.stringify(newStudents));
+  localStorage.setItem("lastStudId", JSON.stringify(newStudId));
+};
+
+// Funzione per eliminare uno studente
+const deleteStudent = (id) => {
+  const foundStudent = getStudent(id);
 
+  if (!!foundStudent) {
+    const prevStudents = getStudentList() || [];
+    const newStudents = prevStudents.filter(({ id: idStudent }) =>
+      idStudent !== "" + id
+    );
+
+    console.log(`student ${JSON.stringify(foundStudent)} deleted.`);
+    localStorage.setItem("students", JSON.stringify(newStudents));
+    return;
+  }
+  console.log(`id ${id} not found in students`);
+};
+
+// Funzione per aggiornare uno studente
+const updateStudent = ({ id, name: newName, lastName: newLastName, email: newEmail, lectures: newLectures }) => {
+  const oldStudent = getStudent(id);
+
+  if (!!oldStudent) {
+    const prevStudents = getStudentList();
+
+    console.log(`updating student with id ${id} : ${JSON.stringify(oldStudent)}`);
+
+    const newStudents = prevStudents.map((stud) => {
+      return (stud.id === "" + id)
+        ? {
+          id: "" + id,
+          name: normalizeName(newName) || stud.name,
+          lastName: normalizeName(newLastName) || stud.lastName,
+          email: newEmail || stud.email,
+          lectures: newLectures || stud.lectures
+        }
+        : { ...stud };
```

Comment:
Ma creiamo un nuovo oggetto anzichè restituirlo per l'immutabilità giusto?

Reply:
Yes, I liked this style to show it gets the same properties